### PR TITLE
:sparkles: Feature/move kubeconfig functionality to util

### PR DESCRIPTION
**Pull request by kashifest**
**What this PR does / why we need it**:

This PR makes available the functionality which was previously in clusterclient for getting and extracting a kubeconfig for given cluster, also in util/kubeconfig/kubeconfig.go.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1306

### CODE REVIEW

File: util/kubeconfig/kubeconfig.go [hunk 1]
```diff
@@ -37,17 +37,15 @@ var (
 	ErrSecretMissingValue = errors.New("missing value in secret")
 )
 
-// SecretName generates the expected name for the KubeConfig Secret to access
```
Comment (maelk) [inline util/kubeconfig/kubeconfig.go]: why remove this comment ?

File: util/kubeconfig/kubeconfig.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ import (
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
```
Comment (maelk) [inline util/kubeconfig/kubeconfig.go]: no need for this anymore

File: util/kubeconfig/kubeconfig.go [hunk 3]
```diff
@@ -37,17 +37,15 @@ var (
...
 
-// SecretName generates the expected name for the KubeConfig Secret to access
-// a remote cluster given the cluster's name.
 func SecretName(cluster string) string {
 	return fmt.Sprintf("%s-kubeconfig", cluster)
 }
 
 // GetSecret retrieves the KubeConfig Secret (if any) from the given
 // cluster name and namespace.
-func GetSecret(c client.Client, cluster, namespace string) (*corev1.Secret, error) {
```
Comment (maelk) [inline util/kubeconfig/kubeconfig.go]: client could remain as is since you removed the struct

File: util/kubeconfig/kubeconfig.go (hunk 1 above)
Comment (kashifest) [reply util/kubeconfig/kubeconfig.go]: This was a mistake.

**Pull request merged**